neo4j/transaction_config.go

```diff
@@ -29,6 +32,8 @@ type TransactionConfig struct {
 	Timeout time.Duration
 	// Metadata is the configured transaction metadata that will be attached to the underlying transaction.
 	Metadata map[string]interface{}
+	// Context is the context to govern the execution
+	Context context.Context
```

Comment:
Support for context is a different story and needs to go a lot deeper. I agree that it is needed but it is misplaced in this context (pun intended).

Reply:
It is the least invasive version, not breaking the API. Happy for any better way to get it down there :) A different way may solve the second of your questions, too :)

Reply:
Ok, since I didn't see the consuming side of the instrumentation it wasn't obvious that you used context to store data between the before/after hooks.
I really don't want to add Context just for implementing a placeholder for instrumentation state, it will look like the driver supports contexts in general which it currently doesn't.
Makes more sense to support context first and then add any instrumenation on top of that if context should be part of the instrumentation api.